trie/secure_trie.go

```diff
@@ -36,32 +42,32 @@ import (
 //
 // SecureTrie is not safe for concurrent use.
 type SecureTrie struct {
-	trie             Trie
-	hashKeyBuf       [common.HashLength]byte
-	secKeyCache      map[string][]byte
-	secKeyCacheOwner *SecureTrie // Pointer to self, replace the key cache on mismatch
-}
-
-// NewSecure creates a trie with an existing root node from a backing database
-// and optional intermediate in-memory node pool.
-//
-// If root is the zero hash or the sha3 hash of an empty string, the
-// trie is initially empty. Otherwise, New will panic if db is nil
-// and returns MissingNodeError if the root node cannot be found.
-//
-// Accessing the trie loads nodes from the database or node pool on demand.
-// Loaded nodes are kept around until their 'cache generation' expires.
-// A new cache generation is created by each call to Commit.
-// cachelimit sets the number of past cache generations to keep.
-func NewSecure(root common.Hash, db *Database) (*SecureTrie, error) {
-	if db == nil {
-		panic("trie.NewSecure called without a database")
+	tree *MerkleTree
+}
+
+var mtDbOnce sync.Once
+var mtDb db.Storage
+
+// NewSecure creates a trie
+func NewSecure(root common.Hash, _ *Database) (*SecureTrie, error) {
```

Comment:
Why not use db by parameter?

Reply:
1. iden3 mekletree needs a raw leveldb.
2. create at leveldb at here is a temporary workaround